### PR TITLE
[ENH] Raise `AssertionError` when `raw_corrs` contains at least one NaN.

### PR DESCRIPTION
**Pull request by JulioAPeraza**
Closes None.

Changes proposed in this pull request:

- Raise `AssertionError` when `raw_corrs` contains at least one NaN.

### CODE REVIEW

**Comment by tsalo** [comment]
Beyond NaNs in the time series, can any other conditions lead to NaNs in the correlations?

**Comment by JulioAPeraza** [comment]
In my case, one ROI (250) fell outside the brain mask, so its time series was zero. This resulted in a zero variance and thus in a correlation equal to infinity. I think we can check both the time series and the variance within ROIs (see below). @tsalo  What would you recommend?

``` python
if np.any(np.isnan(raw_ts)):
    raise ValueError(f"Time series of {files[i_subj]} contains NaNs")

for i, sub_raw_ts in enumerate(raw_ts):
    sub_variance = np.var(sub_raw_ts)
    if sub_variance == 0.0:
        raise ValueError(
            f"ROI {i} with var {sub_variance}, correlation coefficients for"
            f" {files[i_subj]} will be replaced by NaNs"
        )
```

**Comment by tsalo** [comment]
That sounds great!

File: ddmra/workflows.py [hunk 1]
```diff
@@ -119,6 +119,17 @@ def run_analyses(
...
+        if np.any(np.isnan(raw_ts)):
+            raise ValueError(f"Time series of {files[i_subj]} contains NaNs")
+
+        for i, sub_raw_ts in enumerate(raw_ts):
+            sub_variance = np.var(sub_raw_ts)
+            if sub_variance == 0.0:
+                raise ValueError(
+                    f"ROI {i} with var={sub_variance}, correlation coefficients for"
+                    f" {files[i_subj]} will be replaced by NaNs"
+                )
```
[the remark this answers is not in the document]
Comment (JulioAPeraza) [reply ddmra/workflows.py:131]: Thanks for the review and suggestions. I tested this on my data and it worked with `axis=1`.

**Comment by tsalo** [comment]
I think this is good to merge. Do you want to add yourself to the Zenodo file before we do that though?

**Comment by JulioAPeraza** [comment]
Thanks @tsalo

**Pull request merged**